tests/Microsoft.Identity.Web.Test/DownstreamWebApiSupport/ExtraParametersTests.cs

```diff
@@ -0,0 +1,236 @@
+using Xunit;
+
+namespace Microsoft.Identity.Web.Tests
+{
+    public class ExtraParametersTests
+    {
+        private readonly IAuthorizationHeaderProvider _authorizationHeaderProvider;
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly IOptionsMonitor<DownstreamApiOptions> _namedDownstreamApiOptions;
+        private readonly ILogger<DownstreamApi> _logger;
+        private readonly DownstreamApi _downstreamApi;
+
+        public ExtraParametersTests()
+        {
+            _authorizationHeaderProvider = new MyAuthorizationHeaderProvider();
+            _httpClientFactory = new HttpClientFactoryTest();
+            _namedDownstreamApiOptions = new MyMonitor();
+            _logger = new LoggerFactory().CreateLogger<DownstreamApi>();
+
+            _downstreamApi = new DownstreamApi(
+                _authorizationHeaderProvider,
+                _namedDownstreamApiOptions,
+                _httpClientFactory,
+                _logger);
+        }
+
+        [Fact]
+        public async Task UpdateRequestAsync_WithExtraHeaderParameters_AddsHeadersToRequest()
+        {
+            // Arrange
+            var httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, "https://example.com/api");
+            var options = new TestDownstreamApiOptions()
+            {
+                ExtraHeaderParameters = new Dictionary<string, string>
+                {
+                    { "OData-Version", "4.0" },
+                    { "Custom-Header", "test-value" }
+                }
+            };
+
+            // Act
+            await _downstreamApi.UpdateRequestAsync(httpRequestMessage, null, options, false, null, CancellationToken.None);
+
+            // Assert
+            Assert.True(httpRequestMessage.Headers.Contains("OData-Version"));
+            Assert.True(httpRequestMessage.Headers.Contains("Custom-Header"));
+            Assert.Equal("4.0", httpRequestMessage.Headers.GetValues("OData-Version").First());
+            Assert.Equal("test-value", httpRequestMessage.Headers.GetValues("Custom-Header").First());
+        }
+
+        [Fact]
+        public async Task UpdateRequestAsync_WithExtraQueryParameters_AddsQueryParametersToUrl()
+        {
+            // Arrange
+            var httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, "https://example.com/api");
+            var options = new TestDownstreamApiOptions()
+            {
+                ExtraQueryParameters = new Dictionary<string, string>
+                {
+                    { "param1", "value1" },
+                    { "param2", "value2" }
+                }
+            };
+
+            // Act
+            await _downstreamApi.UpdateRequestAsync(httpRequestMessage, null, options, false, null, CancellationToken.None);
+
+            // Assert
+            var requestUri = httpRequestMessage.RequestUri!.ToString();
+            Assert.Contains("param1=value1", requestUri);
```

Comment:
For the Assert.Contains where you have a string, you'd want to use the override with a StringComparison